jctools-core/src/main/java/org/jctools/queues/atomic/BaseLinkedAtomicQueue.java

```diff
@@ -53,6 +53,51 @@ protected final void spConsumerNode(LinkedQueueAtomicNode<E> node) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method is only safe to call from the consumer thread!
+     */
+    @Override
+    public final boolean remove(Object o) {
+        if (null == o) {
+            throw new NullPointerException();
+        }
+
+        for (LinkedQueueAtomicNode<E> prevConsumerNode = null, currConsumerNode = getNextConsumerNode(lpConsumerNode());
```

Comment:
There wasn't any examples of iteration that I could find but the idea was to start `current` at a "real" node (that contains a value) and maintain `prev`/`next` to do the removal. `current` should always represent "real" node during iteration and is safe to compare the value. Do you have an alternative suggestion? I'm also fine if you want to re-work after the PR is pulled in too.